priv/priv.go

```diff
@@ -36,7 +36,8 @@ var (
 
 // Mutex is the snappy mutual exclusion primitive.
 type Mutex struct {
-	lock *FileLock
+	filename string
+	flock    LockedFile
```

Comment:
It feels like we have two different APIs for doing exactly the same thing: one keeps the filename internally, the other doesn't. Can we unify them into a single implementation?